x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyBoolQueryBuilder.java

```diff
@@ -36,10 +39,14 @@ public class ApiKeyBoolQueryBuilder extends BoolQueryBuilder {
         "_id",
         "doc_type",
         "name",
+        "type",
+        API_KEY_TYPE_RUNTIME_MAPPING_FIELD,
         "api_key_invalidated",
         "invalidation_time",
         "creation_time",
-        "expiration_time"
+        "expiration_time",
+        "creator.principal",
+        "creator.realm"
```

Comment:
I slipped in this tiny unrelated change.
It explicitly mentions `creator.principal` and `creator.realm` as fields allowed to be referred to in queries, rather than the current "anything that starts with `creator.`" approach. I think this slightly simplifies the understanding of it all (only `metadata.*` is allowed in a wildcard fashion), since only these two fields can be handled by `ApiKeyFieldNameTranslators#FIELD_NAME_TRANSLATORS` and anything else errors.